extensions/microprofile-metrics/runtime/src/main/java/org/apache/camel/quarkus/component/microprofile/metrics/runtime/CamelMicroProfileMetricsRecorder.java

```diff
@@ -61,4 +63,24 @@ public void configureCamelContext(CamelMicroProfileMetricsConfig config,
             managementStrategy.addEventNotifier(new MicroProfileMetricsCamelContextEventNotifier());
         }
     }
+
+    private static class MicroProfileMetricsContextConfigurerListener extends MainListenerSupport {
+        private static final Logger LOGGER = Logger.getLogger(MicroProfileMetricsContextConfigurerListener.class);
+        private final CamelMicroProfileMetricsConfig config;
+
+        public MicroProfileMetricsContextConfigurerListener(CamelMicroProfileMetricsConfig config) {
+            this.config = config;
+        }
+
+        @Override
+        public void configure(CamelContext camelContext) {
+            if (camelContext.isMessageHistory() && config.enableMessageHistory) {
```

Comment:
I wonder if we can simplify this to just:

```
camelContext.setMessageHistory(true);
camelContext.setMessageHistoryFactory(new MicroProfileMetricsMessageHistoryFactory());
```

And not bother with the other checks? 

In 3.1.0, when you add the `MessageHistoryFactory`, message history is [auto enabled](https://github.com/apache/camel/blob/f984a85ced885e0f8da58d3477b703d9cc2ff88c/core/camel-base/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java#L3887) on the `CamelContext` anyway.

Reply:
It can certainly be done, my only reasoning was: what if the user has explicitly disabled message history ? by adding the extension we would then revert its change without notice.

I guess I can add a log that say that the message history has been turned on in case it was not, maybe the same should happen on 3.1.x.

Reply:
@jamesnetherton I've simplified it a bit, does it look better now ?